hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1169,6 +1180,23 @@ func (r *NodePoolReconciler) listAWSMachineTemplates(nodePool *hyperv1.NodePool)
 	return filtered, nil
 }
 
+func (r *NodePoolReconciler) listAgentMachineTemplates(nodePool *hyperv1.NodePool) ([]agentv1.AgentMachineTemplate, error) {
+	agentMachineTemplateList := &agentv1.AgentMachineTemplateList{}
+	if err := r.List(context.Background(), agentMachineTemplateList); err != nil {
+		return nil, fmt.Errorf("failed to list AgentMachineTemplates: %w", err)
+	}
+	filtered := []agentv1.AgentMachineTemplate{}
+	for i, agentMachineTemplate := range agentMachineTemplateList.Items {
+		if agentMachineTemplate.GetAnnotations() != nil {
+			if annotation, ok := agentMachineTemplate.GetAnnotations()[nodePoolAnnotation]; ok &&
+				annotation == client.ObjectKeyFromObject(nodePool).String() {
```

Comment:
Let's set back this annotation here https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/nodepool/nodepool_controller.go#L1316

Reply:
Done